code/game/objects/items/melee/energy.dm

```diff
@@ -222,6 +222,40 @@
 	else
 		return ..()
 
+/obj/item/melee/transforming/energy/sword/bikehorn
+	name = "bike horn"
+	desc = "A horn off of a bicycle."
+	icon_state = "swordhonk"
+	lefthand_file = 'icons/mob/inhands/weapons/swords_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/swords_righthand.dmi'
+	icon_state_on = "swordpink"
+	saber_color = null
+	light_color = "#ff3399"
+	force = 0
+	throwforce = 0
+	attack_verb = list("HONKED")
+	hitsound = null
+
+/obj/item/melee/transforming/energy/sword/bikehorn/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/squeak, list('sound/items/bikehorn.ogg'=1), 50)
+	qdel(GetComponent(/datum/component/cleave_attack))
```

Comment:
Instead of deleting the component and adding it back every time it's toggled off and on, you should add a trait that hides the examine description of the cleave attack and blocking components.

Reply:
I am too stupid to figure this out which is why I left it for blocking, if you could help out that'd be great.
I removed the cleave attack since I removed the damage when it's in the bikehorn state, I thought it was a bit silly to leave in.